### PR TITLE
Reader type fix 

### LINKED ISSUE #24: Fix Reader Type
ReaderType was implemented here: #23 but I think Reader type needs fixing.
We should read Char by Char and create the lines based on de CSVDefinition line delimiter

### PR DESCRIPTION
**Pull request by ricardochaves9**
this PR fix the bug caught in #24

### CODE REVIEW

File: src/main/scala/parsing/Parser.scala [hunk 1]
```diff
@@ -36,20 +37,19 @@ object Parser extends Encoding {
...
           droppedLines = parsedLines.map(_.merge)
         )
-      case ReaderType(input)   =>
-        val source: Source =  Source.fromString(input.toString)
-        val sourceLines: Iterator[String] = source.getLines()
-        val parsed: Seq[Either[EncodeType, EncodeType]] = sourceLines.toList.map(parsingLogic)
-        source.close()
+      case ReaderType(input) =>
+        val iterator = Iterator.continually(input.read().toChar).toList
+        val lines = iterator.map(_.toString)
```
Comment (joao-penim) [inline src/main/scala/parsing/Parser.scala]: `_.toString` here will make the whole file to be in a single String, it does not separate each line into a separate String

**Review by joao-penim — CHANGES_REQUESTED** [review]

File: src/main/scala/parsing/Parser.scala (hunk 1 above)
Comment (joao-penim) [reply src/main/scala/parsing/Parser.scala]:
```suggestion
        val chars = Iterator.continually(input.read().toChar).toArray
        val source = Source.fromChars(chars)
```
an then replicate Source implementation

Comment (ricardochaves9) [reply src/main/scala/parsing/Parser.scala]: thank you for your suggestion joão , il try to do it

File: src/main/scala/parsing/Parser.scala [hunk 2]
```diff
@@ -36,20 +37,20 @@ object Parser extends Encoding {
...
-      case ReaderType(input)   =>
-        val source: Source =  Source.fromString(input.toString)
-        val sourceLines: Iterator[String] = source.getLines()
-        val parsed: Seq[Either[EncodeType, EncodeType]] = sourceLines.toList.map(parsingLogic)
-        source.close()
+      case ReaderType(input) =>
+        val iterator = Iterator.continually(input.read().toChar).toArray
+        val lines  = Source.fromChars(iterator)
+        lines.close()
+        val parsed = lines.getLines().map(parsingLogic)
```
Comment (joao-penim) [inline src/main/scala/parsing/Parser.scala:43]: close the source only after the getLines()

**Review by joao-penim — APPROVED** [review]
Looks good, just check the comment

File: src/main/scala/parsing/Parser.scala (hunk 2 above)
Comment (ricardochaves9) [reply src/main/scala/parsing/Parser.scala:43]: done , thank you

**Review by joao-penim — APPROVED** [review]

**Pull request merged**